waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,42 @@
+import type { Request as ExpressRequest } from 'express'
+import type { ProviderId, createUser } from '../../auth/utils.js'
+import { prisma } from '../index.js'
+
+type CommonInput = {
+  hookName: string
+  prisma: typeof prisma
+  req: ExpressRequest
+}
+
+/* On Before Signup Hook */
+export type OnBeforeSignupHookFn = (
```

Comment:
I suggest dropping the `Fn` part, `OnBeforeSignupHook` should be enough.

I know we have `SeverSetupFn` and whatnot, but I'm not a fan of those either. I'd change them to `SetUpServer` or something.

Of course, changing this know makes our public API names inconsistent. If you think keeping the consistency is more important, I'm all for it.

I'd probably sacrifice consistency in this case, but it's your call.